### PR TITLE
Data loading tutorial

### LINKED ISSUE #78: A tutorial on writing custom Datasets + Samplers and using transforms
takeaway from thread: https://discuss.pytorch.org/t/feedback-on-pytorch-for-kaggle-competitions/2252/2

### PR DESCRIPTION
**Pull request by chsasank**
Tutorial on loading datasets.

Here is [faces.zip](https://github.com/pytorch/tutorials/files/1049419/faces.zip) required to run the code.

Fix #78 #7

### CODE REVIEW

**Comment by chsasank** [comment]
Hey @soumith, I'll need your help. Can you please hosts [faces.zip](https://github.com/pytorch/tutorials/files/1049419/faces.zip) at `https://download.pytorch.org/tutorial/faces.zip`?

Would you also please review the tutorial once?

File: beginner_source/data_loading_tutorial.py [hunk 1]
```diff
@@ -0,0 +1,402 @@
...
+# One issue we can see from the above is that the samples are not of the 
+# same size. Most neural networks expect the images of a fixed size.
+# Therefore, we will need to write some prepocessing code.
+# Let's create three transforms:
+#
+# -  ``Rescale``: to scale the image
+# -  ``RandomCrop``: to crop from image randomly. This is data
+#    augmentation.
+# -  ``ToTensor``: to convert the numpy images to torch images (we need to
+#    swap axes).
```
Comment (apaszke) [inline beginner_source/data_loading_tutorial.py]: Isn't it better to use `torchvision` transforms here?

**Comment by chsasank** [comment]
No, because both image and landmarks have to be transformed.

Plus I wanted to show by example, how callable transform classes work.

File: beginner_source/data_loading_tutorial.py (hunk 1 above)
Comment (chsasank) [inline beginner_source/data_loading_tutorial.py]: @apaszke Added an afterword explaining torch vision datasets and transform.

File: beginner_source/data_loading_tutorial.py [hunk 2]
```diff
@@ -0,0 +1,402 @@
...
+          sample_batched['landmarks'].size())
+
+    # observe 4th batch and stop.
+    if i_batch == 3:
+        plt.figure()
+        show_landmarks_batch(sample_batched)
+        plt.axis('off')
+        plt.ioff()
+        plt.show()
+        break
```
Comment (jvmancuso) [inline beginner_source/data_loading_tutorial.py]: This is extremely helpful.  +1 for merging.

**Comment by jvmancuso** [comment]
This is incredibly helpful.  Thanks a bunch, +1 for merging.

**Comment by chsasank** [comment]
@soumith Gentle remainder about hosting the data file.

**Comment by soumith** [comment]
done. tutorial/faces.zip should be now there. sorry for the delay

**Comment by chsasank** [comment]
Thanks a lot!

**Pull request merged**